project_stock/models/project_task.py

```diff
@@ -0,0 +1,241 @@
+        data = self.env["stock.scrap"].read_group(
+            [("task_id", "in", self.ids)], ["task_id"], ["task_id"]
+        )
+        count_data = {item["task_id"][0]: item["task_id_count"] for item in data}
+        for item in self:
+            item.scrap_count = count_data.get(item.id, 0)
+
+    @api.depends("move_ids", "move_ids.state")
+    def _compute_allow_moves_action_confirm(self):
+        for item in self:
+            item.allow_moves_action_confirm = any(
+                move.state == "draft" for move in item.move_ids
+            )
+
+    @api.depends("move_ids", "move_ids.state")
+    def _compute_allow_moves_action_assign(self):
+        for item in self:
+            item.allow_moves_action_assign = any(
+                move.state in ("confirmed", "partially_available")
+                for move in item.move_ids
+            )
+
+    @api.depends("move_ids", "move_ids.state")
+    def _compute_stock_state(self):
+        for task in self:
+            task.stock_state = "pending"
+            if task.move_ids:
+                states = task.mapped("move_ids.state")
+                for state in ("confirmed", "assigned", "done", "cancel"):
+                    if state in states:
+                        task.stock_state = state
+                        break
+
+    @api.onchange("picking_type_id")
+    def _onchange_picking_type_id(self):
+        self.location_id = self.picking_type_id.default_location_src_id.id
+        self.location_dest_id = self.picking_type_id.default_location_dest_id.id
+
+    def _check_tasks_with_pending_moves(self):
+        if self.move_ids and "assigned" in self.mapped("move_ids.state"):
+            raise UserError(
+                _("It is not possible to change this with reserved movements in tasks.")
+            )
+
+    def _update_moves_info(self):
+        for item in self:
+            item._check_tasks_with_pending_moves()
+            picking_type = item.picking_type_id or item.project_id.picking_type_id
+            location = item.location_id or item.project_id.location_id
+            location_dest = item.location_dest_id or item.project_id.location_dest_id
+            moves = item.move_ids.filtered(
+                lambda x, loc=location, loc_dest=location_dest: (
+                    x.state not in ("cancel", "done")
+                    and (x.location_id != loc or x.location_dest_id != loc_dest)
+                )
+            )
+            moves.update(
+                {
+                    "warehouse_id": location.warehouse_id.id,
+                    "location_id": location.id,
+                    "location_dest_id": location_dest.id,
+                    "picking_type_id": picking_type.id,
+                }
+            )
+        self.action_assign()
+
+    @api.model
+    def _prepare_procurement_group_vals(self):
+        return {"name": "Task-ID: %s" % self.id}
+
+    def action_confirm(self):
+        self.mapped("move_ids")._action_confirm()
```

Comment:
Hey! It'd be cool to trigger _trigger_scheduler in the moves when calling action confirm. It's what the picking does, and it triggers the orderpoints